### PR TITLE
Switching worker indexing feature flag from opt-in to opt-out

### LINKED ISSUE #8847: Removing feature flag for Project Stein
Currently, Project Stein (workerIndexing) is enabled using a feature flag on the Host side `"AzureWebJobsFeatureFlags": "EnableWorkerIndexing" `

Once the new programming model is stabilized, we can plan to remove this feature flag and make workerIndexing as default behavior **per language**

- Python
- Node
- Powershell
- Java

### PR DESCRIPTION
**Pull request by soninaren**
### Issue describing the changes in this PR

resolves https://github.com/Azure/azure-functions-host/issues/8847
Epic - https://github.com/Azure/azure-functions-host/issues/8848

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script/ScriptConstants.cs [hunk 1]
```diff
@@ -118,6 +118,7 @@ public static class ScriptConstants
         public const string FeatureFlagDisableDiagnosticEventLogging = "DisableDiagnosticEventLogging";
         public const string FeatureFlagDisableMergedWebHostScriptHostConfiguration = "DisableMergedConfiguration";
         public const string FeatureFlagEnableWorkerIndexing = "EnableWorkerIndexing";
```
Comment (madelinegordon) [inline src/WebJobs.Script/ScriptConstants.cs]: Would this constant still be needed?

**Review by madelinegordon — APPROVED** [review]

File: src/WebJobs.Script/ScriptConstants.cs (hunk 1 above)
Comment (mathewc) [reply src/WebJobs.Script/ScriptConstants.cs]: Right - we should remove if no longer being used

File: release_notes.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 - Update Java Worker Version to [2.7.0](https://github.com/Azure/azure-functions-java-worker/releases/tag/2.7.0)
 - Update Node.js Worker Version to [3.5.1](https://github.com/Azure/azure-functions-nodejs-worker/releases/tag/v3.5.1)
 - Update ApplicationInsights packages to 2.21.0 (#8838)
+- Making feature flag for worker indxing opt-out instead of opt-in (#8864)
```
Comment (mathewc) [inline release_notes.md]: spelling

**Review by satvu — APPROVED** [review]

**Pull request merged**